_posts/2018-07-25-dota-macos-performance.md

```diff
@@ -0,0 +1,64 @@
+
+#### Experiment Setup
+
+Hardware:
+  - MacBook Pro 2016
+  - 3.3 GHz Intel Core i7 (dual-core)
+  - Intel Iris Graphics 550
+
+Software: macOS 10.14 beta
+Resolution: 1440 x 900
+
+The benchmark scripts can be seen at our [gfx-portability PR](https://github.com/gfx-rs/portability/pull/118). We were running the same scene as [Phoronix](https://www.phoronix.com/scan.php?page=news_item&px=Dota-2-Initial-Mac-Vulkan), just with a longer time frame and slightly adjusted settings.
+
+### Conclusions
+
+MoltenVK does a good job translating Vulkan to Metal. However, benchmarking shows that it has subpar framerate variability (which is not desirable for a competitive game), and can even be slower than OpenGL on low settings. This doesn't match either Phoronix or MoltenVK's numbers, although we could think of the differences in the test setup that could explain that:
+  - _hardware_: ours has 2 physical cores, while Phoronix had 4
+  - _OS_: we are running on 10.14 beta, while Phoronix used 10.13.4
+  - _settings_: we run with `-vulkan_disable_occlusion_queries` since it's not yet implemented in our library
+
+Either way, our benchmarks show that OpenGL is fairly good on MacOS, and it's a high bar to beat for any Vulkan portability implementation, at least in the context of Dota performance. The change of tides on high settings could be explained by the compute shaders use by Dota, which are not available on MacOS's older OpenGL version. In other words, it's not the raw performance to gain on MacOS but rather more features, allowing certain workloads to use the GPU more efficiently.
+
+Rust proved itself viable in complex high-performance systems. We were able to build solid abstractions and hide the complexity behind tiny interfaces, yet reason about low-level performance and optimize it. In the end, we reached and surpassed the effectiveness of a more established native alternative (MoltenVK is written in ObjectiveC and shipped as an official Dota2 DLC), delivering both higher and more stable frame rates.
+
+The Rust ecosystem deserves a special mention: hooking up extra dependencies was a matter of minutes, and most of the time things just worked. Special thanks to [@SSheldon](https://github.com/SSheldon) for macOS API libraries, [@Amanieu](https://github.com/Amanieu) for the faster locks, and of course the Rust team and contributors for making our life so easy. And since we are at it, thanks [@danginsburg](https://github.com/danginsburg) for help in understanding Dota rendering pipeline, [@jrmuizel](https://github.com/jrmuizel) for performance investigation assistance, and of course gfx-rs team of contributors - you are the best!
```

Comment:
'hooking up extra dependencies ~~was~~ *took* a matter of minutes'
'of course *the* gfx-rs team of contributors'